File: 1213. Intersection of Three Sorted Arrays.py
class Solution:
    def arraysIntersection(self, a: List[int], b: List[int], c: List[int]) -> List[int]:
        ans = []
        i = j = k = 0
        while i<len(a) and j<len(b) and k<len(c):
            if a[i]==b[j]==c[k]:
                ans.append(a[i])
                i += 1
                j += 1
                k += 1
            else:
                if a[i]<=b[j] and a[i]<=c[k]: i += 1
                elif b[j]<=a[i] and b[j]<=c[k]: j += 1
                else: k += 1
        return ans

File: 1213. Intersection of Three Sorted Arrays.cpp
class Solution {
public:
    vector<int> arraysIntersection(vector<int>& a, vector<int>& b, vector<int>& c) {
        vector<int> ans;
        int i=0, j=0, k=0;
        while (i<a.size() && j<b.size() && k<c.size()) 
            if (a[i]==b[j] && b[j]==c[k]) {
                ans.push_back(a[i]);
                ++i, ++j, ++k;
            } else if (a[i]<=b[j] && a[i]<=c[k])
                ++i;
            else if (b[j]<=a[i] && b[j]<=c[k])
                ++j;
            else 
                ++k;
        return ans;
    }
};